### DISCUSSION TITLE
suicide by no-branches

### DISCUSSION THREAD

**Discussion by rienafairefr** [trapexit/mergerfs#1073] [General]
Le'ts say you're merging multiple branches by mergerfs, then (accidentally, or not) remove all the branches through the runtime config xattr api, then the mount seems to be borked and unrecoverable.

I tried to re-add branches to the mount, but a setxattr syscall on /.mergerfs first does a getattr on /, which fails because of no branches. 

I'm tempted to think that is not a bug in mergerfs (that's why I didn't put that question in Issues), mergerfs lets me shoot myself in the foot, if I shoot myself in the foot, I'm to blame, not mergerfs. I can just unmount and redo the mount to recover I guess, but mergerfs with no branches is entering into a state that seems to be a complete dead-end, is there a way out ? Maybe mergerfs should competely prevent entering into a no-branches state ?

**Comment by trapexit** [comment]
Not that I've finished it but for mergerfs v3 the whole xattr interface is going away in favor of a toml config file and runtime reload.

That said I could put in a check to error when the result of a request would mean no branches in the current version.
